src/api/entities/HistoricalMultiSigProposal.ts

```diff
@@ -0,0 +1,177 @@
+import BigNumber from 'bignumber.js';
+
+import { Context, Entity } from '~/internal';
+import { multiSigProposalVotesQuery } from '~/middleware/queries';
+import { Query } from '~/middleware/types';
+import { MultiSigProposalVote, SignerType } from '~/types';
+import { Ensured } from '~/types/utils';
+import { middlewareEventDetailsToEventIdentifier, signerValueToSigner } from '~/utils/conversion';
+
+export type MiddlewareMultiSigProposal = {
+  id: string;
+  proposalId: number;
+  multisigId: string;
+  approvalCount: number;
+  rejectionCount: number;
+  creator?: {
+    did: string;
+  };
+  status: string;
+  createdBlockId: string;
+  updatedBlockId: string;
+  datetime: Date;
+};
+
+type MiddlewareMultiSigProposalUniqueIdentifiers = Pick<MiddlewareMultiSigProposal, 'id'>;
+
+export interface HumanReadable {
+  id: string;
+  proposalId: number;
+  multisigId: string;
+  approvalCount: number;
+  rejectionCount: number;
+  creatorDid?: string;
+  status: string;
+  createdBlockId: string;
+  updatedBlockId: string;
+  datetime: string;
+}
+
+enum MultiSigStatusEnum {
+  Pending = 'Pending',
+  Success = 'Success',
+  Failed = 'Failed',
+}
+
+/**
+ * Represents historical MultiSigProposal that no longer exists on chain
+ */
+export class HistoricalMultiSigProposal extends Entity<
+  MiddlewareMultiSigProposalUniqueIdentifiers,
+  HumanReadable
+> {
```

Comment:
IMO this should not be a Entity. A simple interface would do the trick

Reply:
Anyhow other entity related methods would not be used anywhere

Reply:
I still think that is not needed. We would be using the same MultiSigProposal and HistoricalMultiSigProposal to do the exact same things. Also, for consistency, we always return a interface while getting the historical info